app/pods/publications/publication/translations/route.js

```diff
@@ -1,16 +1,86 @@
 import Route from '@ember/routing/route';
+import { action } from '@ember/object';
+import { PAGE_SIZE } from 'frontend-kaleidos/config/config';
+import { tracked } from '@glimmer/tracking';
 
+// use:
+// - isPieceDeletable property
+// - file property avoids error when piece (and file) are deleted
+export class TranslationRow {
+  @tracked piece;
+  @tracked date;
+  @tracked requestActivity;
+
+  // no async constructor() in JS
+  static async create(piece,requestActivity) {
+    const row = new TranslationRow();
+
+    if (piece){
+      row.piece = piece;
+      row.date = piece.created;
+    }
+    if (requestActivity) {
+      row.requestActivity = requestActivity;
+      row.date = requestActivity.startDate;
+    }
+    return row;
+  }
+
+  get isPieceDeletable() {
+    // can be translation or publication related
+    const isUsedInRequest = this.requestActivitiesUsedBy.length > 0;
+    // receivedDate is set if and only if it is a received pieced
+    const isReceived = !!this.piece.receivedDate;
+    const isUsed = isUsedInRequest || isReceived;
+    return !isUsed;
+  }
+}
```

Comment:
Heeft deze class nog een meerwaarde?  In het vorige design was deze class nodig om de tabel met documenten op een eenvoudige manier te kunnen opbouwen in de template, maar aangezien het design nu dichter bij het model aanleunt, zie ik de meerwaarde niet. 
De property `isPieceDeletable` wordt onder vertaling bijvoorbeeld niet meer gebruikt voor zover ik zie.